pyomo/repn/plugins/gams_writer.py

```diff
@@ -201,46 +312,62 @@ def _print_model( self, model, output_file, solver_capability, var_list,
             var = var_list[i]
             i += 1
             v = symbolMap.getObject(var)
-            try:
+            if not v.is_expression():
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
                     if v.bounds == (0,1):
                         binary.append(var)
+                    elif v.lb is not None and v.lb >= 0:
+                        posInts.append(var)
                     else:
-                        ints.append(var)
+                        otherInts.append(var)
                 elif v.lb == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
-            except AttributeError:
-                # This is an Expression object
-                body = StringIO()
-                v.to_string(body)
+            else:
+                # Same domain violation check as above
+                try:
+                    value(v.expr)
+                except:
+                    raise RuntimeError(
+                        "While evaluating: %s\n\tProblemWriter_gams"
+                        " encountered an error. Ensure initial"
+                        " variable values\n\tdo not violate"
+                        " any domains (are you using log/log10?)"
```

Comment:
There are a couple things going on here.  The first is a hack because of how I elected to generate the expression strings for the GAMS writer.  Currently, `Expression` components show up as if they were proper variables in GAMS, which requires them to have valid initial values.

@gseastream: two more things: 
   1. there are other ways to get bogus initial values.  For example: `(-1)**(0.5)`.
   2. why are you doing these checks at all?  Elsewhere in Pyomo, we do not force the initial conditions to be consistent or valid when emitting the model (if AMPL does because of a bad point, that counts as a solver error and not as a writer error).